FpsCheck.py

```diff
@@ -365,31 +372,44 @@ def find_invalid_well_known(self, check_sets):
             # have stored
             try:
                 json_schema = self.open_and_load_json(url)
-                schema_fields = set(self.acceptable_fields) & set(
-                    json_schema.keys())
                 curr_fps_set = check_sets[primary]
-                for field in schema_fields:
-                    if field == "primary":
-                        if json_schema["primary"] != curr_fps_set.primary:
-                            field_sym_difference = [json_schema["primary"], 
-                            curr_fps_set.primary]
-                        else:
-                            field_sym_difference = []
-                    else:
-                        field_sym_difference = set(json_schema[field]) ^ set(
-                        curr_fps_set.relevant_fields_dict[field])
-                        if field == 'ccTLDs':
-                            for aliased_site in json_schema[field]:
-                                field_sym_difference.update(
-                                    set(json_schema[field][aliased_site]) ^ 
-                                    set(
-                                    curr_fps_set.relevant_fields_dict[field]
-                                    [aliased_site]))
-                    if field_sym_difference:
-                        self.error_list.append("The following member(s) of " 
-                        + field + " were not present in both the changelist "
-                        + "and .well-known/first-party-set.json file: " + 
-                        str(sorted(field_sym_difference)))
+                well_known_set = FpsSet(
+                    json_schema.get('ccTLDs'), 
+                    json_schema.get('primary'), 
+                    json_schema.get('associatedSites'), 
+                    json_schema.get('serviceSites'))
+                if well_known_set.primary != curr_fps_set.primary:
+                    field_sym_difference = [well_known_set.primary, curr_fps_set.primary]
+                    self.error_list.append("The following primary was not "
+                    + "present in both the PR and " + WELL_KNOWN + " file: "
+                    + str(sorted(field_sym_difference)))
+                associated_diff = self.check_well_known_list(
+                    "associatedSites",
+                    curr_fps_set.associated_sites, 
+                    well_known_set.associated_sites)
+                self.error_list.extend(associated_diff)
+                service_diff = self.check_well_known_list(
+                    "serviceSites",
+                    curr_fps_set.service_sites, 
+                    well_known_set.service_sites)
+                self.error_list.extend(service_diff)
+                aliased_site_diff = self.check_well_known_list(
+                    "ccTLD key list", 
+                    curr_fps_set.ccTLDs.keys(), 
+                    well_known_set.ccTLDs.keys())
+                self.error_list.extend(aliased_site_diff)
```

Comment:
I was thinking this info would be conveyed by the loop immediately below this, and we didn't have to check the keys separately.